src/app/(customerFacing)/products/page.tsx

```diff
@@ -18,16 +18,32 @@ export default function ProductsPage({
 }: {
   searchParams: { [key: string]: string | string[] | undefined };
 }) {
+
+
+let sortValues  = Object.values(searchParams)
+
+//key map
+const sortKeyMap: { [key: string]: string } = {
+  "new": "createdAt",
+  "asc": "priceInCents",
+  "desc": "priceInCents",
+};
+
+
+  const sortField =  sortKeyMap[sortValues[0] as string]
+  const sortOrder = (sortValues[0]  as "asc" | "desc" | "new") || "desc";
+  const options = { sortField, sortOrder }; // Prepare the object with field and order
```

Comment:
I would extract lines 23-35 into a single function and call it `getSortOptions`. Then save it in a /utils/productFilterHelpers/index.ts